framework/src/main/java/org/checkerframework/framework/type/AnnotatedTypeFactory.java

```diff
@@ -2844,6 +2852,89 @@ public AnnotatedDeclaredType fromNewClass(NewClassTree newClassTree) {
         return type;
     }
 
+    /**
+     * Returns the method parameters for the invoked method (or constructor), with the same number
+     * of arguments as passed to the invocation tree.
+     *
+     * <p>This expands the parameters if the call uses varargs or contracts the parameters if the
+     * call is to an anonymous class that extends a class with an enclosing type. If the call is
+     * neither of these, then the parameters are returned unchanged.
+     *
+     * @param atypeFactory the type factory to use for fetching annotated types
+     * @param method the method or constructor's type
+     * @param args the arguments to the method or constructor invocation
+     * @return a list of the types that the invocation arguments need to be subtype of; has the same
+     *     length as {@code args}
+     */
+    public static List<AnnotatedTypeMirror> adaptParameters(
```

Comment:
What is the point of moving this to the `AnnotatedTypeFactory`? Why make it a static method that takes a ATF as parameter instead of making it an instance method?